api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceServiceImpl.java

```diff
@@ -263,10 +280,15 @@ public Workspace updateWorkspaceAcls(
       } else {
         // This is how to remove a user from the FireCloud ACL:
         // Pass along an update request with NO ACCESS as the given access level.
-        WorkspaceACLUpdate removedUser = new WorkspaceACLUpdate();
-        removedUser.setEmail(currentUserEmail);
-        removedUser = updateFirecloudAclsOnUser(WorkspaceAccessLevel.NO_ACCESS, removedUser);
-        updateACLRequestList.add(removedUser);
+        // Note: do not do for all users group.  Unpublish will pass the specific NO_ACCESS acl
+        // TODO [jacmrob] : have all users pass NO_ACCESS explicitly? Handle filtering on frontend?
+        User foundUser = userDao.findUserByEmail(currentUserEmail);
```

Comment:
First: See below comment - this usage is making me nervous; I think it would be better to avoid this method entirely.

Else: How about a simpler filter of checking the domain? This would not require serial database lookups running in a loop.

Reply:
👍